### PR TITLE
Passing PAT as Org Level Secret

### PR DESCRIPTION
**Pull request by fiddlermikey**

### CODE REVIEW

**Comment by fiddlermikey** [comment]
You may want to wait to approve this PR until after the starter-workflow has been updated.

File: .github/workflows/dotnet-build-and-release.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: Build dotnet solution and potentially upload the release
 on:
-  workflow_call:
+   workflow_call:
```
Comment (doebrowsk) [inline .github/workflows/dotnet-build-and-release.yml:3]: Spacing / indentation can affect the way workflows run sometimes, so you should remove this extra space

File: .github/workflows/dotnet-build-and-release.yml [hunk 2]
```diff
@@ -15,6 +15,11 @@ on:
         description: The relative directory inside the repo where the build artifacts to publish for release will be located
         required: false # leave empty for non-release build
         type: string
+    secrets:
+      token:
+        description: 'Secret token from caller workflow to access private repo'
```
Comment (doebrowsk) [inline .github/workflows/dotnet-build-and-release.yml:20]: The description says "access private repo" but we are trying to grant access to private packages

**Review by doebrowsk — CHANGES_REQUESTED** [review]

File: .github/workflows/dotnet-build-and-release.yml (hunk 1 above)
Comment (fiddlermikey) [reply .github/workflows/dotnet-build-and-release.yml:3]: removed extra space

File: .github/workflows/dotnet-build-and-release.yml (hunk 2 above)
Comment (fiddlermikey) [reply .github/workflows/dotnet-build-and-release.yml]: changed description repo -> packages

**Review by doebrowsk — APPROVED** [review]

**Pull request merged**